design-proposals/eim-modular-decomposition.md

```diff
@@ -0,0 +1,441 @@
+
+### Secure and at scale Device upgrades [Day 2]
+
+- Mutable and immutable OS update along with CVE tracking
+- Secure device de-authorization and decommissioning.
+
+### Third-party integration
+
+- Interfacing with 3rd party device or fleet manager (Lenovo LOCA) enabling Device
+  manageability and Edge AI usecases on 3rd party edge solutions.
+
+## Objectives
+
+Typically, EIM customers fall into three personas:
+
+- Independent Software Vendors (ISVs) or OS Vendors
+- Original Equipment Manufacturers (OEMs)
+- End Customers or Systems Integrators.
+
+Each persona has distinct needs that can be better served through modular consumption of EIM capabilities.
+
+### User stories
+
+Before diving into the proposal, here are some representative user stories that illustrate the need for modular decomposition:
+
+**Independent Software Vendor/Edge solution vendor** can leverage as a reference EIM design and implementation of
+the following workflows
+
+- **Out-of-band Device Management:** As an ISV or edge solution vendor, I want a End-to-end reference solution to
+  automate out-of-band device management using Intel vPRO AMT and ISM, so that I can manage fleets of edge devices
+  efficiently and securely
+
+> **Note:** Out-of-band management has multiple unique usecases and work flows. e.g. device power management,
+> remote KVM access, Boot control, hardware and software inventory etc. Each of these capabilities can be
+> of consumed by customer as an independent workflow.
+
+- **Hardware and Software Observability:** As an ISV or edge solution vendor, I want to access hardware and software
+  observability for Intel Architecture devices, so that higher management stacks can schedule workloads and monitor
+  fleet health using key silicon metrics.
+- **Automated Edge Device Configuration:** As an ISV or edge solution vendor, I want to automate edge device
+  configuration, including BIOS and firmware settings, so that I can achieve zero-touch management of edge devices.
+- **Secure Device Onboarding and OS Provisioning:** As an ISV or edge solution vendor, I want to securely onboard
+  and provision operating systems on edge devices using technologies like HTTPS boot, full-disk encryption, and
+  secure boot, so that device deployments are protected from unauthorized access.
+- **Day-Two Device Lifecycle Management:** As an ISV or edge solution vendor, I want to manage day-two device
+  lifecycle operations, including immutable OS updates, firmware updates, and CVE remediation, so that edge devices
+  remain secure and up-to-date.
+- **Custom Hardware Resource Configuration:** As an ISV or edge solution vendor, I want to customize device
+  configuration for Intel CPU, GPU, and NPU resources during day-one lifecycle management, so that applications can
+  be allocated appropriate hardware resources.
+- **Reference API Integration:** As an ISV or edge solution vendor, I want to use reference APIs for higher-layer
```

Comment:
I'm not sure if this makes sense as a standalone component. All vertical features should have their own "API for higher-level services". Likely, trusted compute should be another vertical feature

Reply:
I see - so maybe data model is better wording than APIs